### DISCUSSION TITLE
MySQL master slave connection string

### DISCUSSION THREAD

**Discussion by harshavmb** [photoprism/photoprism#2895] [General]
Hello,

I'm not sure whether this was already discussed. I searched through the discussions & found the nearest one [here](https://github.com/photoprism/photoprism/issues/460) but I couldn't find `PHOTOPRISM_DATABASE_DSN` in my docker-compose.yml. 

Is it possible to pass connection strings like `PHOTOPRISM_DATABASE_CONN_STR`=`Server=master_server,slave_server;Database=myDataBase;Uid=myUsername;Pwd=myPassword;`?

**Comment by lastzero** [comment]
You can add to your docker-compose.yml whatever you want. It's just an example that does not include all possible variables. See our documentation which config options you can use.

**Reply by harshavmb** [reply]
Sorry for bothering you again!

I did many permutations & combinations of `PHOTOPRISM_DATABASE_DSN` & `PHOTOPRISM_DATABASE_SERVER` but couldn't get it to work. I'm not sure if the `master`-`slave` connection string was ever tried. 

I did look into [gorm](https://gorm.io/docs/connecting_to_the_database.html#MySQL), [connectionstrings](https://www.connectionstrings.com/mysql/) & `stackoverflow` docs but could find mostly `jdbc`/`.net` connection strings. 

Does photoprism support this in its current state? If not supported, I can give it a try with PR.

**Reply by lastzero** [reply]
Check the Go docs, it's not gorm specific. Also only set the DSN and leave the other variables empty.
